lib/ex_webrtc/peer_connection.ex

```diff
@@ -293,13 +302,43 @@ defmodule ExWebRTC.PeerConnection do
   defp apply_remote_description(_type, sdp, state) do
     # TODO apply steps listed in RFC 8829 5.10
     media = hd(sdp.media)
-    {:ice_ufrag, ufrag} = ExSDP.Media.get_attribute(media, :ice_ufrag)
-    {:ice_pwd, pwd} = ExSDP.Media.get_attribute(media, :ice_pwd)
 
-    :ok = ICEAgent.set_remote_credentials(state.ice_agent, ufrag, pwd)
-    :ok = ICEAgent.gather_candidates(state.ice_agent)
+    with {:ice_ufrag, ufrag} <- ExSDP.Media.get_attribute(media, :ice_ufrag),
+         {:ice_pwd, pwd} <- ExSDP.Media.get_attribute(media, :ice_pwd),
+         {:ok, new_transceivers} <- update_transceivers(state.transceivers, sdp) do
+      :ok = ICEAgent.set_remote_credentials(state.ice_agent, ufrag, pwd)
+      :ok = ICEAgent.gather_candidates(state.ice_agent)
+
+      new_remote_tracks =
+        new_transceivers
+        # only take new transceivers that can receive tracks
+        |> Enum.filter(fn tr ->
+          RTPTransceiver.find_by_mid(state.transceivers, tr.mid) == nil and
+            tr.direction in [:recvonly, :sendrecv]
+        end)
+        |> Enum.map(fn tr -> MediaStreamTrack.from_transceiver(tr) end)
+
+      for track <- new_remote_tracks do
+        notify(state.owner, {:track, track})
+      end
```

Comment:
I suppose you know that and this is only temporary, but I believe this notification should appear when only tracks are negotiated.

Reply:
Now we accept everything :D